code/headers/usart_connection.hpp

```diff
@@ -1,52 +1,90 @@
+ * @author Patrick Dekker
+ * @brief base class for a usart implementation.
+ * @version 0.1
+ * @date 2019-05-24
+ *
+ * @copyright Copyright (c) 2019
+ *
+ */
 
 #include <cstdint>
 
 namespace r2d2::usart {
-    /// @brief  base class for usart implementation
+    /**
+     * @brief Base class for the usart implementation
+     *
+     */
     class usart_connection_c {
     public:
-        /// @brief operator << writes byte to usart
-        /// @param byte byte to send
+        /**
+         * @brief operator << writes a byte with the usart
+         *
+         * @param byte to send
+         * @return usart_connection_c&
+         */
         usart_connection_c &operator<<(uint8_t byte);
 
-        /// @brief operator << writes string to usart
-        /// @param c c-style string to send
+        /**
+         * @brief operator << writes a string with the usart
+         *
+         * @param c c-style string to send
+         * @return usart_connection_c&
+         */
         usart_connection_c &operator<<(const char *c);
 
-        /// @brief enables internal USART controler
-        virtual void enable() = 0;
-
-        /// @brief disables internal USART controler
-        virtual void disable() = 0;
-
-        /// @brief send byte using USART
-        /// @param c byte to send
+        /**
+         * @brief Write a uint8_t with the usart
+         *
+         * @param c data to send
+         * @return true if the data is sended correctly
+         * @return false if the data has not been sended correctly
+         */
         virtual bool send(const uint8_t c) = 0;
 
-        /// @brief send char using USART
-        /// @param c char to send
-        virtual void putc(char c) = 0;
-
-        /// @brief receive byte from USART
-        /// @return uint8_t byte received from USART
+        /**
+         * @brief Get a data from the usart
+         *
+         * @return uint8_t
+         */
         virtual uint8_t receive() = 0;
 
-        /// @brief check if char is available
-        /// @return true if byte is available false if not
-        virtual bool char_available() = 0;
+        /**
+         * @brief returns how much data is available
+         *
+         * @return unsigned int amount of bytes
+         */
+        virtual unsigned int available() = 0;
 
-        /// @brief receive char if available
-        /// @return char received
-        virtual char getc() = 0;
+        /**
+         * @brief Write a char with the usart
+         *
+         * @param c
+         */
+        virtual void putc(char c) {
+            send(static_cast<uint8_t>(c));
+        }
 
-        /// @brief returns available data
-        /// @return amount of bytes
-        virtual unsigned int available() = 0;
+        /**
+         * @brief Check if data is available
+         *
+         * @return true if data is available
+         * @return false if no data is available
+         */
+        virtual bool char_available() {
```

Comment:
Could this function be marked `const`?

Reply:
this function is overloaded from istream now and that doesn't have it const